crates/goose-cli/src/commands/acp.rs

```diff
@@ -33,6 +35,116 @@ struct GooseAcpAgent {
+                for content in content_items {
+                    if let RawContent::Text(text_content) = &content.raw {
+                        let text = &text_content.text;
+
+                        // Parse line numbers based on command type and response format
+                        match command {
+                            Some("view") => {
+                                // For view command, look for "lines X-Y" pattern in header
+                                let line = extract_view_line_range(text)
+                                    .map(|range| range.0 as u32)
+                                    .or(Some(1));
+                                locations.push(create_tool_location(path_str, line));
+                            }
+                            Some("str_replace") | Some("insert") => {
+                                // For edits, extract the first line number from the snippet
+                                let line = extract_first_line_number(text)
+                                    .map(|l| l as u32)
+                                    .or(Some(1));
+                                locations.push(create_tool_location(path_str, line));
+                            }
+                            Some("write") => {
+                                // For write, just point to the beginning of the file
+                                locations.push(create_tool_location(path_str, Some(1)));
+                            }
+                            _ => {
+                                // For other commands or unknown, default to line 1
+                                locations.push(create_tool_location(path_str, Some(1)));
+                            }
+                        }
+                        break; // Only process first text content
+                    }
+                }
+            }
+
+            // If we didn't find any locations yet, add a default one
+            if locations.is_empty() {
+                locations.push(create_tool_location(path_str, Some(1)));
+            }
+        }
+    }
+
+    locations
+}
+
+/// Extract line range from view command output (e.g., "### path/to/file.rs (lines 10-20)")
+fn extract_view_line_range(text: &str) -> Option<(usize, usize)> {
+    // Look for pattern like "(lines X-Y)" or "(lines X-end)"
+    let re = regex::Regex::new(r"\(lines (\d+)-(\d+|end)\)").ok()?;
+    if let Some(caps) = re.captures(text) {
+        let start = caps.get(1)?.as_str().parse::<usize>().ok()?;
+        let end = if caps.get(2)?.as_str() == "end" {
+            start // Use start as a reasonable default
+        } else {
+            caps.get(2)?.as_str().parse::<usize>().ok()?
+        };
+        return Some((start, end));
+    }
+    None
+}
+
+/// Extract the first line number from code snippet (e.g., "123: some code")
+fn extract_first_line_number(text: &str) -> Option<usize> {
+    // Look for pattern like "123: " at the start of a line within a code block
```

Comment:
can probably tell goose to drop the commentary, as it is just saying what is in the code (annoys me so much it does this, but I always have to tell it tp rip it out0